test/simulator/timestamp/src/session_simulator.cpp

```diff
@@ -174,37 +174,63 @@ session_simulator::commit_transaction(const std::string &config)
     WT_SIM_RET_MSG(ts_manager->parse_config(config, supported_ops, unsupported_ops, config_map),
       "Incorrect config (" + config + ") passed in commit_transaction");
 
+    uint64_t commit_ts = 0;
+    uint64_t durable_ts = 0;
+
+    if (_has_commit_ts)
+        commit_ts = _commit_ts;
+
+    if (_durable_ts_set)
+        durable_ts = _durable_ts;
+
+    int ret;
     auto pos = config_map.find("commit_timestamp");
     if (pos != config_map.end()) {
-        int ret = ts_manager->validate_hex_value(pos->second, "commit timestamp");
+        ret = ts_manager->validate_hex_value(pos->second, "commit timestamp");
         if (ret != 0) {
             rollback_transaction();
             return (ret);
         }
+        commit_ts = ts_manager->hex_to_decimal(pos->second);
+    }
+
+    pos = config_map.find("durable_timestamp");
+    if (pos != config_map.end()) {
+        ret = ts_manager->validate_hex_value(pos->second, "durable timestamp");
+        if (ret != 0) {
+            rollback_transaction();
+            return (ret);
+        }
+        durable_ts = ts_manager->hex_to_decimal(pos->second);
+    }
 
-        uint64_t commit_ts = ts_manager->hex_to_decimal(pos->second);
+    if (commit_ts != 0) {
         ret = set_commit_timestamp(commit_ts);
         if (ret != 0) {
             rollback_transaction();
             return (ret);
         }
     }
 
-    pos = config_map.find("durable_timestamp");
-    if (pos != config_map.end()) {
+    if (durable_ts != 0) {
+        ret = set_durable_timestamp(durable_ts);
+        if (ret != 0) {
+            rollback_transaction();
+            return (ret);
+        }
     }
 
     if (_prepared_txn) {
         if (!_has_commit_ts)
             WT_SIM_RET_MSG(EINVAL, "commit_timestamp is required for a prepared transaction");
 
-        if (!has_durable_timestamp())
+        if (!is_durable_ts_set())
             WT_SIM_RET_MSG(EINVAL, "durable_timestamp is required for a prepared transaction");
     } else {
         if (has_prepare_timestamp())
             WT_SIM_RET_MSG(EINVAL, "prepare timestamp is set for non-prepared transaction");
 
-        if (has_durable_timestamp())
+        if (is_durable_ts_set())
             WT_SIM_RET_MSG(
               EINVAL, "durable_timestamp should not be specified for non-prepared transaction");
```

Comment:
Np: not sure if we should state the variable name with the underscore in general statements that indicate a rule to follow. For the rule above about prepare timestamp, we don't indicate the variable name.

Reply:
?

Reply:
Updated in [855d63f](https://github.com/wiredtiger/wiredtiger/pull/8398/commits/855d63fd1187a059d930b0d3cac5451796c2a404)